### ISSUE TITLE
Few weaknesses in the code

### ISSUE THREAD

**Issue by AppChecker** [magento/magento2#6715]
Hi!

I hope this will be helpful:

1) 
[app/code/Magento/Usps/Model/Carrier.php](https://github.com/magento/magento2/blob/2c22ac4edddb99383d79a0295858c26ba28bf7b5/app/code/Magento/Usps/Model/Carrier.php#L1908)

```
        if ($response === false || $response->getName() == 'Error') {
            $debugData['result'] = [
                'error' => $response->Description,
                'code' => $response->Number,
                'xml' => $response->asXML(),
            ];
...
        }
```

if $response is false, method asXML will be called on false

2) 
[app/code/Magento/CatalogSearch/Model/Layer/Filter/Decimal.php](https://github.com/magento/magento2/blob/2c22ac4edddb99383d79a0295858c26ba28bf7b5/app/code/Magento/CatalogSearch/Model/Layer/Filter/Decimal.php#L118)

```
$label = $this->renderRangeLabel(
                empty($from) ? 0 : $from,
                empty($to) ? $to : $to
            );
```

Both parts of ternary operators are equals.

3)
[app/code/Magento/CatalogRule/Model/Rule.php](https://github.com/magento/magento2/blob/2c22ac4edddb99383d79a0295858c26ba28bf7b5/app/code/Magento/CatalogRule/Model/Rule.php#L561)

```
if (is_array($value)) {
                    if ($value != $array2[$key]) {
                        $result[$key] = true;
                    }
                } else {
                    if ($value != $array2[$key]) {
                        $result[$key] = true;
                    }
                }
```

Parts of if-conditional operator are equals.

Possible defect was found by Echelon team with [AppChecker static analyzer](https://npo-echelon.ru/en/solutions/appchecker.php)

**Comment by veloraven** [comment]
@AppChecker thank you for your feedback.
Please identify which version of Magento you are running. 
Please create a separate issue for each problem and format them according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines)

**Comment by vzabaznov** [comment]
internal ticket created MAGETWO-62187

**Comment by vzabaznov** [comment]
issue was fixed in fresh magento, thanks for reporting

**Comment by magento-engcom-team** [comment]
Hi @AppChecker. Thank you for your report. 
The issue has been fixed in magento/magento2#21968 by @Nazar65 in 2.3-develop branch
Related commit(s):
 - [7cb0cf8dc21612761b9e58de600447c12ef340dd](https://github.com/magento/magento2/commit/7cb0cf8dc21612761b9e58de600447c12ef340dd)
 - [bbe71ef43062429cd3f235ad885cc3a996e7628d](https://github.com/magento/magento2/commit/bbe71ef43062429cd3f235ad885cc3a996e7628d)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @AppChecker. Thank you for your report. 
The issue has been fixed in magento/magento2#22453 by @Nazar65 in 2.2-develop branch
Related commit(s):
 - [34b2dccf322c1af0d5c563063e439de1b4a9cbe9](https://github.com/magento/magento2/commit/34b2dccf322c1af0d5c563063e439de1b4a9cbe9)

The fix will be available with the upcoming 2.2.9 release.
